src/citrine/informatics/descriptors.py

```diff
@@ -30,8 +31,30 @@ def get_type(cls, data) -> Type[Serializable]:
             "Inorganic": ChemicalFormulaDescriptor,
             "Organic": MolecularStructureDescriptor,
             "Real": RealDescriptor,
+            "Integer": IntegerDescriptor,
         }[data["type"]]
 
+    def _equals(self, other, attrs):
+        """Check to see if the attrs from the other instance match this instance.
+
+        Returns True if all of the attribute names from attrs match between this and
+        the other instance, else False. A missing attribute will also return False.
+
+        Parameters
+        ----------
+        other: Description
+            the Description instance to compare to
+        attrs: List[str]
+            A list of attribute names to lookup and compare
+
+        """
+        try:
+            return all([
+                self.__getattribute__(key) == other.__getattribute__(key) for key in attrs
+            ])
+        except AttributeError:
+            return False
```

Comment:
I realize that this is just abstracted code (good call), but in the case where both objects are missing the same key, it returns False even though the objects are identical.  Not sure how much it matters.

Reply:
I think there's a good solution here.  If any of the `attrs` are missing from `self`, then we should raise an exception, because this is a clear programmer error.  We probably don't want to do this for `other` instance, because this could be any type of class which may not have this attrs.  The unit tests expect `descriptor == None` to be `False` which seems reasonable.